### ISSUE TITLE
Account Settings Email Page Suspense Issue

### ISSUE THREAD

**Issue by ralf-boltshauser** [stack-auth/stack#335]
In the Account Settings handler, when switching to the email tab, the nav is gone because a suspense boundary is triggered on level too high.

**Comment by N2D4** [comment]
Do you want to work on this? @ralf-boltshauser

**Comment by ralf-boltshauser** [comment]
yes sure, currently I am actually debugging some bigger issue that actually might be related to stack-auth and be the fundament of this. I will come back later and maybe delete / merge this issue into the larger thing. 

I figured out, that in stack-auth projects (need to confirm) when I execute a server action the loading.tsx is retriggered. Whereas in normal nextjs applications the loading.tsx only triggered in the first render and then afterwards it just updates the state. 

As soon as I have narrowed down that thing I will update the issue and take a look at it!

**Comment by fomalhautb** [comment]
> yes sure, currently I am actually debugging some bigger issue that actually might be related to stack-auth and be the fundament of this. I will come back later and maybe delete / merge this issue into the larger thing.
> 
> I figured out, that in stack-auth projects (need to confirm) when I execute a server action the loading.tsx is retriggered. Whereas in normal nextjs applications the loading.tsx only triggered in the first render and then afterwards it just updates the state.
> 
> As soon as I have narrowed down that thing I will update the issue and take a look at it!

This might be because of this https://github.com/vercel/next.js/issues/50163

Next.js 14 automatically refreshes the page when setting cookies in server actions. However, based on my testing, Next.js 15 doesn't seem to have this problem. So, if it is not too hard, you can upgrade to Next.js 15, and the problem should be solved. We will probably try to come up with something to workaround this, like disabling cookies in server action.

**Comment by ralf-boltshauser** [comment]
ahh I see thank you, I will give it a shot!
